app/scripts/HorizontalLine1DPixiTrack.js

```diff
@@ -216,6 +229,43 @@ class HorizontalLine1DPixiTrack extends HorizontalTiled1DPixiTrack {
     this.refreshTiles();
 
     this.draw();
+
+    const isValueScaleLocked = this.isValueScaleLocked();
+
+    if (
+      this.continuousScaling &&
+      this.minValue() !== undefined &&
+      this.maxValue() !== undefined &&
+      this.valueScaleMin === null &&
+      this.valueScaleMax === null &&
+      !isValueScaleLocked
+    ) {
+      const newMin = this.minVisibleValue();
+      const newMax = this.maxVisibleValue();
+
+      const epsilon = 1e-6;
+
+      if (
+        newMin !== null &&
+        newMax !== null &&
+        (Math.abs(this.minValue() - newMin) > epsilon ||
+          Math.abs(this.maxValue() - newMax) > epsilon)
+      ) {
+        this.minValue(newMin);
+        this.maxValue(newMax);
+
+        this.scheduleRerender();
+      }
+    }
+
+    if (
+      this.continuousScaling &&
+      isValueScaleLocked &&
+      this.minValue() !== undefined &&
+      this.maxValue() !== undefined
```

Comment:
Three of these conditions are the same as in the previous `if` condition. Those conditions could be precomputed or the two conditions could be nested to avoid code duplication.

Reply:
Done